lib/new_relic/telemetry/plug.ex

```diff
@@ -0,0 +1,222 @@
+defmodule NewRelic.Telemetry.Plug do
```

Comment:
Oh, curious: does this need to be a gen_server? I assume that some of things it is calling need a process as the caller or something like that. I don't think telemetry strictly requires it though, just a function, maybe? Or there's likely some other reason.

Reply:
This is similar to the other PR, I realized.